pkg/operator/ceph/cluster/controller.go

```diff
@@ -362,30 +365,31 @@ func (c *ClusterController) configureLocalCephCluster(namespace, name string, cl
 				}
 				return false, nil
 			}
-
-			c.updateClusterStatus(clusterObj.Namespace, clusterObj.Name, cephv1.ClusterStateCreating, "")
+			message := config.CheckConditionReady(c.context, namespace, name)
+			config.ConditionExport(c.context, namespace, name,
+				cephv1.ConditionProgressing, v1.ConditionTrue, "ClusterProgressing", message)
 
 			err = cluster.createInstance(c.rookImage, *cephVersion)
 			if err != nil {
 				failedMessage = fmt.Sprintf("failed to create cluster in namespace %q. %v", cluster.Namespace, err)
 				logger.Errorf(failedMessage)
 				return false, nil
 			}
-
-			state = cephv1.ClusterStateCreated
+			config.ConditionExport(c.context, namespace, name,
+				cephv1.ConditionReady, v1.ConditionTrue, "ClusterCreated", "Cluster created successfully")
 			failedMessage = ""
 			return true, nil
 		})
 
 	if err != nil {
+		config.ConditionExport(c.context, namespace, name,
+			cephv1.ConditionFailure, v1.ConditionTrue, "ClusterFailure", "Giving up waiting for cluster creating")
 		return errors.Wrapf(err, "giving up waiting for cluster creating")
 	}
 
-	c.updateClusterStatus(clusterObj.Namespace, clusterObj.Name, state, failedMessage)
-
-	if state == cephv1.ClusterStateError {
-		// the cluster could not be initialized
-		return errors.New("failed to initialized the cluster")
+	msg := config.ErrorMapping()
+	if msg != nil {
+		return msg
```

Comment:
always name your errors, if we return we should explain what was the reason.

`return errors.Wrapf(msg, "failed to ...")`

Assuming `msg` is a type error.